runtime/pkg/metricssql/parser.go

```diff
@@ -0,0 +1,607 @@
+func (c *Compiler) Compile(ctx context.Context, sql string) (string, string, []*runtimev1.ResourceName, error) {
+	stmtNodes, _, err := c.p.ParseSQL(sql)
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	if len(stmtNodes) != 1 {
+		return "", "", nil, errors.New("metrics sql: expected exactly one SQL statement")
+	}
+
+	stmt, ok := stmtNodes[0].(*ast.SelectStmt)
+	if !ok {
+		return "", "", nil, errors.New("metrics sql: expected a SELECT statement")
+	}
+
+	t := &transformer{
+		controller:       c.controller,
+		instanceID:       c.instanceID,
+		userAttributes:   c.userAttributes,
+		dimExprToCol:     make(map[string]string),
+		measureExprToCol: make(map[string]string),
+	}
+	compiledSQL, err := t.transformSelectStmt(ctx, stmt)
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	return compiledSQL, t.connector, t.refs, nil
+}
+
+type transformer struct {
+	controller     *runtime.Controller
+	instanceID     string
+	userAttributes map[string]any
+
+	metricsView   *runtimev1.MetricsViewV2
+	refs          []*runtimev1.ResourceName
+	connector     string
+	dimsToExpr    map[string]string
+	measureToExpr map[string]string
+
+	// resolved dimension expression to dimension name
+	dimExprToCol map[string]string
+	// resolved measure expression to measure name
+	measureExprToCol map[string]string
+}
+
+func (t *transformer) transformSelectStmt(ctx context.Context, node *ast.SelectStmt) (string, error) {
+	if node.WithBeforeBraces {
+		return "", fmt.Errorf("metrics sql: WITH clause is not supported")
+	}
+	if node.IsInBraces {
+		return "", fmt.Errorf("metrics sql: sub select is not supported")
+	}
+	if node.With != nil {
+		return "", fmt.Errorf("metrics sql: WITH clause is not supported")
+	}
+
+	// parse from clause
+	if node.From.TableRefs == nil {
+		return "", fmt.Errorf("metrics sql: need from clause")
+	}
+
+	var sb strings.Builder
+	sb.WriteString("SELECT ")
+	fromClause, err := t.transformFromClause(ctx, node.From)
+	if err != nil {
+		return "", err
+	}
+
+	selectList, err := t.transformSelectStmtColumns(ctx, node)
+	if err != nil {
+		return "", err
+	}
+
+	sb.WriteString(selectList)
+	sb.WriteString(" FROM ")
+	sb.WriteString(fromClause)
+
+	if node.Where != nil {
+		where, err := t.transformExprNode(ctx, node.Where, false)
+		if err != nil {
+			return "", err
+		}
+		sb.WriteString(" WHERE ")
+		sb.WriteString(where)
+	}
+	if len(t.measureExprToCol) > 0 && len(t.dimExprToCol) > 0 {
+		if node.GroupBy != nil {
+			return "", fmt.Errorf("metrics sql: group by clause is implicitly added when any measure is selected. The implicit group by includes all selected dimensions")
+		}
+
+		sb.WriteString(" GROUP BY ")
+		dimExprList := maps.Keys(t.dimExprToCol)
+		// sort so that consistent group by sql is produced
+		slices.Sort(dimExprList)
+		for i, expr := range dimExprList {
+			if i != 0 {
+				sb.WriteString(", ")
+			}
+			sb.WriteString(expr)
+		}
+	} // todo :: support group by if only dimensions are selected
```

Comment:
If possible, it would be nice to support this now for both dimensions and measures. Having just a dimension is an easy way to get distinct dimension values. Having just a measure is equivalent to a total query.

Reply:
So it is possible to specify just dimension or just measure now also. This is specifically for cases when we want to select just dimensions and group by on one or more dimensions. I can handle that case as well.

Reply:
Okay I realised this can not be achieved till we allow some aggregate functions over dimensions. 
So now we can already do : `select dim1, dim2 from mv` or `select measure1 from mv` but not `select sum(dim1), dim2 from mv group by dim2`. 
I will also added support for `select distinct col1 [,col2...] from mv` type query as well.
Any other queries that you have in your mind ?

Reply:
Got it – we don't need (maybe even don't want) to support aggregations over dimensions, so that should be fine